pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -649,7 +649,7 @@ func (p *criStatsProvider) getAndUpdateContainerUsageNanoCores(stats *runtimeapi
 		defer p.mutex.Unlock()
 
 		cached, ok := p.cpuUsageCache[id]
-		if !ok || cached.stats.UsageCoreNanoSeconds == nil {
+		if !ok || cached.stats.UsageCoreNanoSeconds == nil || stats.Cpu.UsageCoreNanoSeconds.Value < cached.stats.UsageCoreNanoSeconds.Value {
```

Comment:
Can we return an error to match what we do for negative time values?

Reply:
My one concern with making this an error is currently this condition will also get triggered by the bug addressed in https://github.com/kubernetes/kubernetes/pull/83504. Though I suppose an error message for hitting this case would have aided me in debugging what was going on when I was hitting that bug.

Reply:
Correct me if I am wrong, but it is a bug if cpu usage reported is reset for a running container.  We should fail loudly if at all possible, rather than fail silently.  If there are cases where it is expected to be reset, we should log a warning.